CMakeLists.txt

```diff
@@ -149,9 +149,19 @@ if(${Acts_VERSION} VERSION_LESS ${Acts_VERSION_MIN}
         AND NOT "${Acts_VERSION}" STREQUAL "9.9.9")
     message(FATAL_ERROR "Acts version ${Acts_VERSION_MIN} or higher required, but ${Acts_VERSION} found")
 endif()
+# ActsCore
+get_target_property(ACTS_COMPILE_FEATURES ActsCore INTERFACE_COMPILE_FEATURES)
+if (NOT "cxx_std_${CMAKE_CXX_STANDARD}" IN_LIST ACTS_COMPILE_FEATURES)
+  message(FATAL_ERROR "Acts has not been built with the requested C++ standard.")
```

Comment:
What is the worst that can happen if we make this a non-fatal error? There are some practical cases when compiling with differing standards may be a fine escape hatch. I see folks are adding such checks to their projects, but don't get what is the actual goal. And I'm not aware of any ABI incompatibility between the recent standards.

Reply:
There are cases where C++20 support will cause the implementation of different interfaces to be activated, compared to C++17 support. When trying to link against it in the C++17 standard, you might end up with undefined references.

But realistically, I think the worst that can happen is probably just that warnings go ignored for too long...

Maybe a warning is enough?

Reply:
I would keep it a warning. I think, the API inconsistencies, that you mention, if present, will show up loudly during the compilation.